commcare_connect/users/views.py

```diff
@@ -44,3 +49,22 @@ def get_redirect_url(self):
 
 
 user_redirect_view = UserRedirectView.as_view()
+
+
+class CreateUserLinkView(ClientProtectedResourceMixin, APIView):
```

Comment:
This mixin enforces OAuth client authentication, so it will only be accessible to systems like hq that have a client id and client secret.

Reply:
One, sort of weird, alternative here is that HQ could forward the token from the user linking the account (effectively impersonating the connect user). It makes the auth a little less scary (client secret would no longer allow full access to this view), but is a bit gross having hq impersonate users by forwarding on their credentials.

Reply:
+1 for having separate auth for HQ and not impersonating the user